frontend/src/pages/projects/projectSharing/utils.ts

```diff
@@ -0,0 +1,38 @@
+import { RoleBindingKind } from '~/k8sTypes';
+import { DashboardConfig } from '~/types';
+import { featureFlagEnabled } from '~/utilities/utils';
+import { getDisplayNameFromK8sResource } from '~/pages/projects/utils';
+import { ProjectSharingRBType, ProjectSharingRoleType } from './types';
+
+export const isProjectSharingEnabled = (dashboardConfig: DashboardConfig) =>
+  featureFlagEnabled(dashboardConfig.spec.dashboardConfig.disableProjectSharing);
+
+export const getRoleBindingResourceName = (roleBinding: RoleBindingKind): string =>
+  roleBinding.metadata?.name || '';
+export const getRoleBindingDisplayName = (roleBinding: RoleBindingKind): string =>
+  getDisplayNameFromK8sResource(roleBinding);
+
+export const filterRoleBindingSubjects = (
+  roleBindings: RoleBindingKind[],
+  type: ProjectSharingRBType,
+): RoleBindingKind[] => roleBindings.filter((roles) => roles.subjects[0]?.kind === type);
+
+export const castProjectSharingRoleType = (role: string): ProjectSharingRoleType | undefined =>
+  ProjectSharingRoleType[role.toUpperCase() as keyof typeof ProjectSharingRoleType];
```

Comment:
This is bad TS... if you really need to convert the role into ProjectType which I am not sure you even need to do... support checking if `role` is in `ProjectSharingRoleType` in a type guard ([TS documentation](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates))

`as` is a naughty keyword... you should avoid using it. Anytime you want to use it, you're likely trying to cut corners on the right solution. 

If you have to do this (for time reasons)... wouldn't this work?

```suggestion
  ProjectSharingRoleType[role.toUpperCase() as ProjectSharingRoleType];
```

I highly encourage you to find alternatives to this... this feels wrong from a principle point of view.

Reply:
Not even needed, refactored to the two values we have now.